### PR TITLE
4.0

### PR DESCRIPTION
**Pull request by ihorrusinko**
- update botkit to version 4.0
- bind Botmock core with botkit
- add new examples and tests for slack
- update docs

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Change Log
 
-# 0.7.4
+### 4.0.0
+Major update with breaking changes
```
Comment (amplicity) [inline CHANGELOG.md]: If we put this in master, how will users designate prior botkit versions for mock to use?

File: README.md [hunk 2]
```diff
@@ -57,35 +71,68 @@ it('should return `help message` if user types `help`', () => {
...
 });
 ```
-## Advanced Usage ##
 
-`botExtender` - allows developers to extend the bot and add custom functions to the `bot`. This overrides functionality in [BotmockWorker.js](https://github.com/gratifychat/botkit-mock/blob/migrate_to_botkit_core/lib/BotmockWorker.js).
+## Advance usage
+`Botkit` itself depends on adapters (slack, facebook, etc.).
+`Botmock` is an extension of `Botkit` provides interface for accepting user messages through `.usersInput`.
+You could connect any botkit providers to botmock.
+For now, we have only advance slack binding for the [slack provider](https://github.com/howdyai/botkit/tree/master/packages/botbuilder-adapter-slack).
```
Comment (amplicity) [inline README.md]: What file is this mocked in?

Comment (ihorrusinko) [reply README.md]: There is no actual file with full slack mock as in a previous `botkit-mock`, with a new botkit structure we use a package `botbuilder-adapter-slack` it is standalone solution.
Generally we have `SlackApiMock` that helps us to bind only mocked api calls to `slack api` over `botbuilder-adapter-slack` so we do not implement by ourself bot methods.

File: CHANGELOG.md (hunk 1 above)
Comment (ihorrusinko) [reply CHANGELOG.md]: it is gonna be the same version naming as for botkit

**Pull request merged**